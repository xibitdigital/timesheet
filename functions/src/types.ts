export interface HolidayDayType {
  date: string
  localName: string
  name: string
  countryCode: string
  fixed: boolean
  global: boolean
  counties: string[]
  launchYear?: number
  type: string
}

export interface DayType {
  date: string
  type: string
}

export interface DatesReqQueryType {
  year: string
  month: string
  countryCode: string
  timeSheetId: string
  clientId: string
}

export interface WorkDay {
  dayType: string
  date: string
  clientId: string
  timeSheetId: string
  workedHours: number
  owner: string
}

export interface TimeSheet {
  name: string
  clientId: string
  projectId: string
  month: string
  year: string
  owner: string
  countryCode: string
}

export enum DayFlag {
  PUBLIC = 'Public',
  WEEKEND = 'Weekend',
  WEEKDAY = 'Weekday',
}
