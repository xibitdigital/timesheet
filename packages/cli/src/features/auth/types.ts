// TYPES
export type Profile = {
  authenticated: boolean
  idToken: number
  profile: any
  expiresAt: string
}

// STATE
export type AuthState = {
  readonly data: Profile
  readonly loading: boolean
  readonly errors?: string
}
