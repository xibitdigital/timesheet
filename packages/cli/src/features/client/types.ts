// TYPES
export type Client = {
  name: string
  address: string
  postcode: string
  vat: string
}

// STATE
export type ClientState = {
  readonly data: Client[]
  readonly loading: boolean
  readonly updating: boolean
  readonly errors?: string
}
