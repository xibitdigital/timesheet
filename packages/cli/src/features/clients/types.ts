// TYPES
export type Client = {
  name: string
  address: string
  postcode: string
  vat: string
}

// ACTION TYPES
export enum Actions {
  FETCH_LIST_REQUEST = 'FETCH_LIST_REQUEST',
  FETCH_LIST_SUCCESS = 'FETCH_LIST_SUCCESS',
  FETCH_LIST_ERROR = 'FETCH_LIST_ERROR',
}

type FetchClientsRequestAction = {
  type: Actions.FETCH_LIST_REQUEST
}

type FetchClientsSuccessAction = {
  type: Actions.FETCH_LIST_SUCCESS
  payload: Client[]
}

type FetchClientsErrorAction = {
  type: Actions.FETCH_LIST_ERROR
  payload: string
}

export type ClientsActionTypes =
  | FetchClientsRequestAction
  | FetchClientsSuccessAction
  | FetchClientsErrorAction

// STATE
export type ClientsState = {
  readonly data: Client[]
  readonly loading: boolean
  readonly updating: boolean
  readonly errors?: string
}
