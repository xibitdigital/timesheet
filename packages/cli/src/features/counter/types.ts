import { Actions } from './actionTypes'

interface IncrementCounterAction {
  type: Actions.INCREMENT_COUNTER
}
interface DecrementCounterAction {
  type: Actions.DECREMENT_COUNTER
}
export type CounterActionTypes = IncrementCounterAction | DecrementCounterAction

export interface SystemState {
  count: {
    value: number
  }
}
