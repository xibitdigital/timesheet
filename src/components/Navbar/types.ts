import { Routes } from '../../shared/routes'

export type RouteAction = () => void

export interface NavBarMenuItem {
  title: string
  action: RouteAction
  icon: React.ReactNode
}
