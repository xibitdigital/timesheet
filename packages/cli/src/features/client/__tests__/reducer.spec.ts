import clientReducer from '../reducer'
import { ClientState } from '../types'

/**
 * FIXTURES
 */
const getInitialState = (initial?: Partial<ClientState>) =>
  clientReducer(initial as ClientState, {} as any)

/**
 * STORIES
 */
describe('AuditDetails Stories', () => {
  const initialState = getInitialState()

  describe('initial state', () => {
    it('should match a snapshot', () => {
      expect(initialState).toMatchSnapshot()
      expect(initialState.data.length).toEqual(0)
      expect(initialState.errors).toEqual(undefined)
      expect(initialState.loading).toEqual(false)
      expect(initialState.updating).toEqual(false)
    })
  })

  describe('fetchList', () => {})

  describe('error', () => {})
})
