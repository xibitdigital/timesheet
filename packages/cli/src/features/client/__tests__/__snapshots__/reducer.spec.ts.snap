// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuditDetails Stories initial state should match a snapshot 1`] = `
Object {
  "data": Array [],
  "errors": undefined,
  "loading": false,
  "updating": false,
}
`;
